examples/webhook.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/xanzy/go-gitlab"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+// parse errors
+var (
+	errInvalidHTTPMethod             = errors.New("invalid HTTP Method")
+	errMissingGitLabEventHeader      = errors.New("missing X-Gitlab-Event Header")
+	errGitLabTokenVerificationFailed = errors.New("X-Gitlab-Token validation failed")
+	errEventNotFound                 = errors.New("event not defined to be parsed")
+	errParsingPayload                = errors.New("error parsing payload")
+)
+
+// webhookExample shows how to create a Webhook server to parse Gitlab events
+// listens /webhook endpoint on host 0.0.0.0 port 8080
+func webhookExample() {
+	wh := webhook{
+		Secret: "your-gitlab-secret",
+		// only listen for MergeRequest and Pipeline events
+		EventsToAccept: []gitlab.EventType{gitlab.EventTypeMergeRequest, gitlab.EventTypePipeline},
+	}
+
+	mux := http.NewServeMux()
+	mux.Handle("/webhook", wh)
+	_ = http.ListenAndServe("0.0.0.0:8080", mux)
```

Comment:
Please handle the errors (goes for the whole file).

Reply:
hey, I am adding a fatal log in case the HTTP server start fails. 

how would you like me to handle all the other errors like writing a response to the HTTP request, closing the request body, etc?

Reply:
I would say that in an example like this, using `log.Fatalf` or just `log.Fatal` should be fine 👍🏻